frontend/src/metabase/visualizations/visualizations/LinkViz/LinkViz.tsx

```diff
@@ -1,76 +1,153 @@
-  isSettings,
   onUpdateVisualizationSettings,
   settings,
 }: LinkVizProps) {
   const {
-    link: { url },
+    link: { url, entity },
   } = settings;
 
   const isNew = !!dashcard?.justAdded;
+  const [autoFocus, setAutoFocus] = useState(isNew);
+  const previousUrl = usePrevious(url);
+
+  const handleLinkChange = (newLink: string) =>
+    onUpdateVisualizationSettings({ link: { url: newLink } });
 
-  const handleLinkChange = (e: React.ChangeEvent<HTMLInputElement>) =>
-    onUpdateVisualizationSettings({ link: { url: e.currentTarget.value } });
+  const handleEntitySelect = (entity: LinkEntity) => {
+    onUpdateVisualizationSettings({
+      link: {
+        entity: {
+          id: entity.id,
+          db_id: entity.model === "table" ? entity.database_id : undefined,
+          name: entity.name,
+          model: entity.model,
+          description: entity.description,
+          display: entity.display,
+        },
+      },
+    });
+  };
 
-  const showEditBox = isEditing && !isPreviewing && !isSettings;
+  const [inputIsFocused, { turnOn: onFocusInput, turnOff: onBlurInput }] =
+    useToggle();
+
+  useEffect(() => {
+    // if the url was auto-filled from the entity, focus the input
+    if (previousUrl === undefined && !!url) {
+      setAutoFocus(true);
+    }
+  }, [previousUrl, url]);
+
+  if (entity) {
+    const wrappedEntity = Search.wrapEntity({
+      ...entity,
+      database_id: entity.db_id ?? entity.database_id,
+      table_id: entity.model === "table" ? entity.id : undefined,
+      collection: {},
+    });
+
+    if (isEditing) {
+      return (
+        <EditLinkCardWrapper>
+          <EntityDisplay entity={wrappedEntity} showDescription={false} />
+        </EditLinkCardWrapper>
+      );
+    }
+
+    return (
+      <DisplayLinkCardWrapper>
+        <CardLink to={wrappedEntity.getUrl()} target="_blank" rel="noreferrer">
+          <EntityDisplay entity={wrappedEntity} showDescription />
+        </CardLink>
+      </DisplayLinkCardWrapper>
+    );
+  }
 
-  if (showEditBox) {
+  if (isEditing) {
     return (
       <EditLinkCardWrapper>
-        <Input
-          autoFocus={isNew}
-          placeholder={"https://example.com"}
-          value={url ?? ""}
-          onChange={handleLinkChange}
-          fullWidth
-          // the dashcard really wants to turn all mouse events into drag events
-          onMouseDown={e => e.stopPropagation()}
-        />
+        <TippyPopover
+          visible={!!url?.length && inputIsFocused && !isUrlString(url)}
+          content={
+            <SearchResultsContainer>
+              <SearchResults
+                searchText={url?.trim()}
+                onEntitySelect={handleEntitySelect}
+              />
+            </SearchResultsContainer>
+          }
+          placement="bottom"
+        >
+          <Input
+            fullWidth
+            value={url ?? ""}
+            autoFocus={autoFocus}
+            placeholder={t`https://example.com`}
```

Comment:
I believe this shouldn't be translatable

Reply:
http://www.exemplo.com/ ?